ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/merging/StartupMergingProcessor.java

```diff
@@ -51,6 +51,7 @@ protected void handleAnnotations(final DeploymentUnit deploymentUnit, final EEAp
             if (data != null) {
                 if (!data.getClassLevelAnnotations().isEmpty()) {
                     description.initOnStartup();
+                    description.getModuleDescription().registerStartupBean();
```

Comment:
Why does this only handle annotations, and not the deployment descriptor as well?